presto-main/src/test/java/io/prestosql/sql/planner/TestLogicalPlanner.java

```diff
@@ -1274,4 +1274,18 @@ public void testRedundantHashRemovalForUnionAllAndMarkDistinct()
                                                                         node(ProjectNode.class,
                                                                                 node(TableScanNode.class))))))))));
     }
+
+    @Test
+    public void testRemoveRedundantCrossJoin()
+    {
+        assertPlan("WITH temp AS (SELECT 1 AS a) SELECT regionkey from nation, temp WHERE regionkey = temp.a",
```

Comment:
add a test case with `SELECT *` -- the `1` from the scalar subquery should be pulled up and cross join still eliminated.

Reply:
But in case `SELECT * ` -- then we might not be able to pull up now. We need something like a `EqualityInference` for ProjectExpressions which I guess we don't have rn.

Reply:
Right. You can add a commented out test case, with `TODO https://github.com/prestosql/presto/issues/2480` -- https://github.com/prestosql/presto/issues/2480.